src/main/java/racingcar/logic/RacingCarService.java

```diff
@@ -0,0 +1,76 @@
+package racingcar.logic;
+
+import camp.nextstep.edu.missionutils.Randoms;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import racingcar.validator.CarNameValidator;
+
+public class RacingCarService {
+
+    CarNameValidator carNameValidator = new CarNameValidator();
+
+    public HashMap<String, Integer> saveCars(String carList) {
+
+        List<String> cars = List.of(carList.split(",", -1)); // 공백도 허용
+
+        HashMap<String, Integer> result = new HashMap<>();
+
+        for (String car : cars) {
+            carNameValidator.validateCarName(car);
+            carNameValidator.checkDuplicateCarName(result, car);
+            result.put(car, 0);
+        }
+
+        return result;
+    }
+
+    public HashMap<String, Integer> saveResult(HashMap<String, Integer> result) {
+
+        for (String car : result.keySet()) {
+            int randomNum = extractRandom();
+
+            if (randomNum >= 4) {
+                int afterRaceNum = result.get(car) + 1;
+                result.put(car, afterRaceNum);
+            }
```

Comment:
```suggestion
            if (randomNum >= 4) {
                result.replace(car, result.get(car) + 1);
            }
```
이러한 방식도 있어요!
물론 둘다 동일한 로직이지만 의미 전달 측면에서는 조금 더 쉽게 이해할 수 있을 것 같아요

Reply:
오.. HashMap에서 key값이 존재하는 상태에서 put을 하면 키가 대체된다고 해서 put으로 했는데, 말씀대로 replace가 더 의미에 맞는 메서드인 것 같네요!